torch/csrc/autograd/python_function.cpp

```diff
@@ -929,6 +929,21 @@ PyObject* THPFunction_register_hook(PyObject* _self, PyObject* hook) {
   END_HANDLE_TH_ERRORS
 }
 
+PyObject* THPFunction_register_prehook(PyObject* _self, PyObject* hook) {
+  HANDLE_TH_ERRORS
+  auto self = (THPFunction*)_self;
+  auto cdata = self->cdata.lock();
+  TORCH_CHECK(
+      cdata,
+      "Attribute 'register_prehook' is invalid for this instance of _C._FunctionBase. "
```

Comment:
Didn't we remove the python code for the old style custom Function? :p

Reply:
I guess.. but people can still do `Func().name()` lol